---
title: "CELLO: Cancer EvoLutionary analysis using LOngitudinal genomic data"
author: "SONG, Dong"
date: " 5/26/2020"
output: github_document
---

```{r setup, global_options, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.path='../Output/', warning=FALSE, message=FALSE)
```

## Ownership
Wang Lab at HKUST (http://wang-lab.ust.hk/)

## Introduction
Cancer EvoLutionary analysis using LOngitudinal genomic data (CELLO) is a protocal for comprehensive analysis of longitudinal genomic sequencing data in cancer. It was originally developed by Jiguang Wang [1,2], and was then packed up by Biaobin Jiang and Dong Song in MATLAB and R seperately. This code was written in R.

## Datasets
The input SAVI report (input.savi.txt) consists of a list of genetic variants from 90 glioblastoma patients.

## Loading CELLO package
```{r}
source('CELLO.R')
```

## CELLO Pipeline
```{r}
savi.table <- mutRead("../../input.savi.txt",20,1,5)
```

```{r}
head(savi.table[,1:10])
```


```{r}
knownDriverGene <- c('LTBP4','PTPN11','NF1','RB1','PDGFRA','PIK3CG','PIK3R1','PIK3CA','PTEN','EGFR','IDH1','ATRX','TP53')
stats <- mutStats(savi.table, knownDriverGene,5,remove_LOW = TRUE)
```

```{r}
head(stats$mutNum.table)
```

```{r}
head(stats$mutGenes.table)
```

```{r fig.asp = 0.65, fig.width = 12, fig.align = 'center' ,dpi=300}
mutLandscape(stats$mutNum.table,stats$mutGenes.table)
```

```{r fig.asp = 1, fig.width = 10, fig.align = 'center',dpi=300}
mutCorrelation(stats$mutGenes.table)
```

```{r fig.asp = 0.7, fig.width = 10, fig.align = 'center',dpi=300}
freq.table <- mutFrequency(savi.table, knownDriverGene,stats$mutGenes.table,5)
head(freq.table)
```

```{r fig.asp = 0.35, fig.width = 15, fig.align = 'center',dpi=300}
hm.table <- mutSignature(savi.table,15,350,1.2)
```

```{r fig.asp = 0.7, fig.width = 10, fig.align = 'center',dpi=300}
cluster.table <- mutTreeClustering(stats$mutNum.table)
```

```{r fig.asp = 0.7, fig.width = 12, fig.align = 'center',dpi=300}
switch.table <- mutSwitch(savi.table,knownDriverGene,5,20)
```


```{r fig.asp = 0.7, fig.width = 12, fig.align = 'center',dpi=300}
selGene <-c('LTBP4','IDH1','ATRX','TP53','NF1','MSH6','PIK3CG','PIK3R1','PIK3CA','PTEN','EGFR')
selStats <- mutStats(savi.table, selGene,5,remove_LOW = TRUE)
TEDG <- mutDirectedGraph(selStats$mutGenes.table)
```
 
```{r}
TEDG$node.table
```
 
```{r}
TEDG$edge.table
```


## Reference
[1] Wang, J., Cazzato, E., Ladewig, E., Frattini, V., Rosenbloom, D. I., Zairis, S., ... & Lee, J. K. (2016). Clonal evolution of glioblastoma under therapy. Nature Genetics, 48(7), 768-776.

[2] Wang, J., Khiabanian, H., Rossi, D., Fabbri, G., Gattei, V., Forconi, F., ... & Pasqualucci, L. (2014). Tumor evolutionary directed graphs and the history of chronic lymphocytic leukemia. Elife, 3, e02869.